src/geo_traits_impl/geometry.rs

```diff
@@ -0,0 +1,364 @@
+            None => panic!("GeoJSON feature has no geometry"),
+        }
+    }
+}
+
+impl geo_traits::GeometryTrait for &crate::Feature {
+    type T = f64;
+    type PointType<'b> = PointType where Self: 'b;
+    type LineStringType<'b> = LineStringType where Self: 'b;
+    type PolygonType<'b> = PolygonType where Self: 'b;
+    type MultiPointType<'b> = MultiPointType where Self: 'b;
+    type MultiLineStringType<'b> = MultiLineStringType where Self: 'b;
+    type MultiPolygonType<'b> = MultiPolygonType where Self: 'b;
+    type GeometryCollectionType<'b> = GeometryCollectionType where Self: 'b;
+    type RectType<'b> = UnimplementedRect<Self::T> where Self: 'b;
+    type TriangleType<'b> = UnimplementedTriangle<Self::T> where Self: 'b;
+    type LineType<'b> = UnimplementedLine<Self::T> where Self: 'b;
+
+    fn as_type(
+        &self,
+    ) -> geo_traits::GeometryType<
+        '_,
+        Self::PointType<'_>,
+        Self::LineStringType<'_>,
+        Self::PolygonType<'_>,
+        Self::MultiPointType<'_>,
+        Self::MultiLineStringType<'_>,
+        Self::MultiPolygonType<'_>,
+        Self::GeometryCollectionType<'_>,
+        Self::RectType<'_>,
+        Self::TriangleType<'_>,
+        Self::LineType<'_>,
+    > {
+        crate::Feature::as_type(self)
+    }
+
+    fn dim(&self) -> Dimensions {
+        crate::Feature::dim(self)
+    }
+}
+
+impl geo_traits::GeometryTrait for crate::GeoJson {
+    type T = f64;
+    type PointType<'b> = PointType where Self: 'b;
+    type LineStringType<'b> = LineStringType where Self: 'b;
+    type PolygonType<'b> = PolygonType where Self: 'b;
+    type MultiPointType<'b> = MultiPointType where Self: 'b;
+    type MultiLineStringType<'b> = MultiLineStringType where Self: 'b;
+    type MultiPolygonType<'b> = MultiPolygonType where Self: 'b;
+    type GeometryCollectionType<'b> = GeometryCollectionType where Self: 'b;
+    type RectType<'b> = UnimplementedRect<Self::T> where Self: 'b;
+    type TriangleType<'b> = UnimplementedTriangle<Self::T> where Self: 'b;
+    type LineType<'b> = UnimplementedLine<Self::T> where Self: 'b;
+
+    fn dim(&self) -> Dimensions {
+        match self {
+            crate::GeoJson::Feature(f) => f.dim(),
+            crate::GeoJson::FeatureCollection(fc) => fc.dim(),
+            crate::GeoJson::Geometry(g) => g.dim(),
+        }
+    }
+
+    fn as_type(
+        &self,
+    ) -> geo_traits::GeometryType<
+        '_,
+        Self::PointType<'_>,
+        Self::LineStringType<'_>,
+        Self::PolygonType<'_>,
+        Self::MultiPointType<'_>,
+        Self::MultiLineStringType<'_>,
+        Self::MultiPolygonType<'_>,
+        Self::GeometryCollectionType<'_>,
+        Self::RectType<'_>,
+        Self::TriangleType<'_>,
+        Self::LineType<'_>,
+    > {
+        match self {
+            crate::GeoJson::Feature(f) => f.as_type(),
+            crate::GeoJson::FeatureCollection(_fc) => {
+                unimplemented!("TODO")
```

Comment:
TODO here needs to be completed

Reply:
I assume a `FeatureCollection`, interpreted as a geometry-trait, would map to a `GeometryCollection`?

Reply:
Yep exactly. `FeatureCollection.as_type` is already implemented to return a `GeometryCollection`. We just need to add that detail here in `GeoJson::FeatureCollection`.